sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -398,133 +384,117 @@ export class EventHubSender extends LinkEntity {
     const retryOptions = options.retryOptions || {};
     const timeoutInMs = getRetryAttemptTimeoutInMs(retryOptions);
     retryOptions.timeoutInMs = timeoutInMs;
-    const sendEventPromise = (): Promise<void> =>
-      new Promise<void>(async (resolve, reject) => {
-        const rejectOnAbort = (): void => {
-          const desc: string =
-            `[${this._context.connectionId}] The send operation on the Sender "${this.name}" with ` +
-            `address "${this.address}" has been cancelled by the user.`;
-          // Cancellation is user-intended, so log to info instead of warning.
-          logger.info(desc);
-          return reject(new AbortError("The send operation has been cancelled by the user."));
-        };
-
-        if (abortSignal && abortSignal.aborted) {
-          // operation has been cancelled, so exit quickly
-          return rejectOnAbort();
-        }
-
-        const removeListeners = (): void => {
-          clearTimeout(waitTimer); // eslint-disable-line @typescript-eslint/no-use-before-define
-          if (abortSignal) {
-            abortSignal.removeEventListener("abort", onAborted); // eslint-disable-line @typescript-eslint/no-use-before-define
-          }
-        };
-
-        const onAborted = (): void => {
-          removeListeners();
-          return rejectOnAbort();
-        };
-
-        if (abortSignal) {
-          abortSignal.addEventListener("abort", onAborted);
+    const sendEventPromise = async (): Promise<void> => {
+      const initStartTime = Date.now();
+      if (!this.isOpen()) {
+        const senderOptions = this._createSenderOptions(timeoutInMs);
+        try {
+          await waitForTimeoutOrAbortOrResolve({
+            actionFn: () => {
+              return defaultLock.acquire(this.senderLock, () => {
+                return this._init(senderOptions);
+              });
+            },
+            abortSignal: options?.abortSignal,
+            timeoutMs: timeoutInMs,
+            timeoutMessage:
+              `[${this._context.connectionId}] Sender "${this.name}" ` +
+              `with address "${this.address}", was not able to send the message right now, due ` +
+              `to operation timeout.`
+          });
+        } catch (err) {
+          const translatedError = translate(err);
+          logger.warning(
+            "[%s] An error occurred while creating the sender %s: %s",
+            this._context.connectionId,
+            this.name,
+            `${translatedError?.name}: ${translatedError?.message}`
+          );
+          logErrorStackTrace(translatedError);
+          throw translatedError;
```

Comment:
Not sure this catch is needed as you've pointed out above `retry` will handle this.

Reply:
yes, it is gone, now that I have pulled it out into common code to be shared with getMaxMessageSize() in https://github.com/Azure/azure-sdk-for-js/pull/13826/commits/ec7f720c474b9973d4f14ebe6388b5fda98dd680